src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDashboard/requestLog.tsx

```diff
@@ -0,0 +1,331 @@
+  currentPage: number;
+  requests: SentryAppWebhookRequest[];
+};
+
+export default class RequestLog extends AsyncComponent<Props, State> {
+  shouldReload = true;
+
+  get eventTypes() {
+    // TODO(nola): ideally this would be kept in sync with EXTENDED_VALID_EVENTS on the backend
+
+    const {app} = this.props;
+    let events = [ALL_EVENTS, 'installation.created', 'installation.deleted'];
+
+    if (app.events.includes('error')) {
+      events = [...events, 'error.created'];
+    }
+    if (app.events.includes('issue')) {
+      events = [
+        ...events,
+        'issue.created',
+        'issue.resolved',
+        'issue.ignored',
+        'issue.assigned',
+      ];
+    }
+    if (app.isAlertable) {
+      events = [...events, 'event_alert.triggered'];
+    }
+
+    const issueLinkComponent: any = (app.schema.elements || []).find(
+      (element: any) => element.type === 'issue-link'
+    );
+    if (issueLinkComponent) {
+      const issueLinkEvents = ['external_issue.created', 'external_issue.linked'];
+      if (
+        (issueLinkComponent.create && issueLinkComponent.create.uri) ||
+        (issueLinkComponent.link && issueLinkComponent.link.uri)
+      ) {
+        issueLinkEvents.push('select_options.requested');
+      }
+      events = [...events, ...issueLinkEvents];
+    }
+
+    return events;
+  }
+
+  get hasNextPage() {
+    return (this.state.currentPage + 1) * MAX_PER_PAGE < this.state.requests.length;
+  }
+
+  get hasPrevPage() {
+    return this.state.currentPage > 0;
+  }
+
+  getEndpoints(): Array<[string, string, any] | [string, string]> {
+    const {slug} = this.props.app;
+
+    const query: any = {};
+    if (this.state) {
+      if (this.state.eventType !== ALL_EVENTS) {
+        query.eventType = this.state.eventType;
+      }
+      if (this.state.errorsOnly) {
+        query.errorsOnly = true;
+      }
+    }
+
+    return [['requests', `/sentry-apps/${slug}/requests/`, {query}]];
+  }
+
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      requests: [],
+      eventType: ALL_EVENTS,
+      errorsOnly: false,
+      currentPage: 0,
+    };
+  }
+
+  onChangeEventType = eventType => {
+    this.setState({
+      eventType,
+      currentPage: 0,
+    });
+    this.remountComponent();
+  };
+
+  onChangeErrorsOnly = () => {
+    this.setState({
+      errorsOnly: !this.state.errorsOnly,
+      currentPage: 0,
+    });
+    this.remountComponent();
+  };
+
+  onNextPage = () => {
+    this.setState({
+      currentPage: this.state.currentPage + 1,
+    });
+  };
+
+  onPrevPage = () => {
+    this.setState({
+      currentPage: this.state.currentPage - 1,
+    });
+  };
+
+  renderLoading() {
+    return this.renderBody();
+  }
+
+  renderBody() {
+    const {requests, eventType, errorsOnly, currentPage} = this.state;
+    const {app} = this.props;
+
+    const currentRequests = requests.slice(
+      currentPage * MAX_PER_PAGE,
+      (currentPage + 1) * MAX_PER_PAGE
```

Comment:
@nolachen why wouldn't we do the pagination at the request level instead of doing it at the component level? We are loading a bunch of items we aren't rendering.

Reply:
@scefali it wouldn't be the same or as simple as other endpoints because we're using redis instead of django models, and there is a max of 100*2*(# of events) items that we're loading, however i can look into paginating them if you think it's worth it

Reply:
@nolachen got it, probably not worth the effort then